### ISSUE TITLE
Introduce DI container

### ISSUE THREAD

**Issue by TimoHeiten** [TimoHeiten/zer0mqXt#10]
LiteInjector, SimpleInjector or anything of that sorts

**Comment by Flash0ver** [comment]
Should we also support `Microsoft.Extensions.DependencyInjection`, for which there are already many extension-packages?

**Comment by TimoHeiten** [comment]
Yeah maybe we should start with that one instead. Not sure for now though.
My initial idea was that you hook up the IBus/ISocket in MS DI yourself. But an extension method in Microsoft.Extensions.DependencyInjection namespace would not harm and is easy to do.

LiteInjector would be only used internally in the library, and we could then provide some configuration methods or callbacks.
